src/wallet/api/wallet.cpp

```diff
@@ -632,12 +632,12 @@ void WalletImpl::refreshThreadFunc()
     LOG_PRINT_L3(__FUNCTION__ << ": starting refresh thread");
 
     while (true) {
-        std::unique_lock<std::mutex> lock(m_refreshMutex);
+        boost::mutex::scoped_lock lock(m_refreshMutex);
         if (m_refreshThreadDone) {
             break;
         }
         LOG_PRINT_L3(__FUNCTION__ << ": waiting for refresh...");
-        m_refreshCV.wait_for(lock, std::chrono::seconds(m_refreshIntervalSeconds));
+        m_refreshCV.wait(lock);
```

Comment:
I recall some dodgy code here, not sure that will actually work, need pointing out to mbg033 to make sure